scm/github.go

```diff
@@ -7,17 +7,35 @@ import (
 	"strings"
 
 	"github.com/gabrie30/ghorg/configs"
-	"github.com/gabrie30/ghorg/internal/repo"
 	"github.com/google/go-github/v32/github"
 	"golang.org/x/oauth2"
 )
 
+var (
+	_ Client = GithubClient{}
+)
+
+func init() {
+	registerClient(GithubClient{})
+}
+
+type GithubClient struct {
```

Comment:
Might also make more sense for this to just be called Github{} since it had a field called client already. Also do you think all the providers, gitlab, gitea, etc should also follow this pattern?

Then the function signatures would be a bit more clear. Although I think setting the client should ideally be done out of this function, maybe in a NewGitHub function where the client is passed in? That way you can inject the client for testing purposes. What do you think?

```
func (gh Github) GetOrgRepos(targetOrg string) ([]Repo, error) {
    gh.client = gh.newClient()
    ....
}

```

Reply:
It depends on how mouch this pull should refactor at once ...

Reply:
... lot of things are stored into env variables, if we go this way we should remove it pice by pice (and only leave it as alternative to flags ...)

Reply:
Sounds good, thanks again for the PR. Ghorg isn't as testable as it should be, which will be the next priority, I think this is a step in the right direction.